system-contracts/contracts/NonceHolder.sol

```diff
@@ -132,12 +132,13 @@ contract NonceHolder is INonceHolder, ISystemContract {
     /// @notice Increments the deployment nonce for the account and returns the previous one.
     /// @param _address The address of the account which to return the deploy nonce for.
     /// @return prevDeploymentNonce The deployment nonce at the time this function is called.
-    function incrementDeploymentNonce(address _address) external returns (uint256 prevDeploymentNonce) {
+    function incrementDeploymentNonce(address _address) external onlySystemCall returns (uint256 prevDeploymentNonce) {
```

Comment:
Maybe it is a bad idea. Probably we should change a require below instead

Reply:
With the change now using createEVM, that modifier is not needed anymore.